### PR TITLE
Rewritten add_path and pylint corrections

### PR DESCRIPTION
**Pull request by OneTorcha**

### CODE REVIEW

File: src/crl/rfcli/rfcli.py [hunk 1]
```diff
@@ -201,8 +198,6 @@ def use_jybot(self):
         if self.rfcli_args.enable_jybot:
             if RobotCommand.is_jybot_installed():
                 return True
-            else:
-                raise Exception("Jybot is not installed.")
```
Comment (vereszol) [inline src/crl/rfcli/rfcli.py]: Could you explain the reason to remove this?

File: src/crl/rfcli/rfcli.py [hunk 2]
```diff
@@ -223,8 +218,8 @@ def run(self):
             return command.execute()
         finally:
             if self.output_under_public_html:
-                print('HTML logs might be located at: http://%s/~%s/rfcli/log.html' % (
-                    socket.getfqdn(), self._user_dir_property()))
+                print(f"HTML logs might be located at: http://%s/~%s/rfcli/log.html"
+                      f" {socket.getfqdn(), self._user_dir_property()}")
```
Comment (vereszol) [inline src/crl/rfcli/rfcli.py:227]: The resulting printout is not the same as before.

File: src/crl/rfcli/rfcli.py [hunk 3]
```diff
@@ -268,27 +262,35 @@ def _get_environment(self):
...
+        return variables
+
     def get_variables(self):
         options = []
-        try:
-            self.parser.readfp(open(self.absfilename))  # pylint:disable=deprecated-method
-        except Exception as e:
-            raise Exception("Cannot open target ini file %s: %s" % (self.absfilename, e))
-        if not self.parser.has_section('target'):
-            raise Exception("Target ini file %s does not have the [target] section" % self.absfilename)
```
Comment (vereszol) [inline src/crl/rfcli/rfcli.py]: I think the same checks should be done and same exceptions should be raised (or some subclass of Exception)

File: src/crl/rfcli/rfcli.py (hunk 1 above)
Comment (OneTorcha) [reply src/crl/rfcli/rfcli.py]: Misunderstood the pylint. 

 ```
def use_jybot(self):
        if self.rfcli_args.enable_jybot:
            if RobotCommand.is_jybot_installed():
                return True
            raise ValueError("Jybot is not installed.")
        return False
```

File: src/crl/rfcli/rfcli.py (hunk 2 above)
Comment (OneTorcha) [reply src/crl/rfcli/rfcli.py:227]: print(f"HTML logs might be located at: http://{socket.getfqdn()}/~{self._user_dir_property()}/rfcli/log.html")

File: src/crl/rfcli/rfcli.py (hunk 3 above)
Comment (OneTorcha) [reply src/crl/rfcli/rfcli.py]: I get same errors without those exections:

robot run-test: commands[1] | rfcli --statusrc --target testi9 -i koe testcases/entity-test/file_hunter/file_hunter.robot
/work/targets/testi9.ini
NoSectionError: No section: 'target'

robot run-test: commands[1] | rfcli --statusrc --target test9 -i koe testcases/entity-test/file_hunter/file_hunter.robot
ValueError: Target file test9 does not exist

**Comment by theodore86** [comment]
@OneTorcha do you need help on this one? Looking forward to get the latest release

**Comment by OneTorcha** [comment]
@theodore86 maybe, review could be nice and for some reason sourcing libs and resources need "./" after this change..

like libraries/own_lib.py needs to be modified ../libraries/own_lib.py

Just havent had time to figure out or is that a big problem even.

**Comment by theodore86** [comment]
@OneTorcha thanks, i will get a look at it

File: src/crl/rfcli/rfcli.py (hunk 3 above)
Comment (vereszol) [reply src/crl/rfcli/rfcli.py]: For a human user these might be the same errors, but programmatically they are different.

Exception type change (NoSectionError and ValueError vs the previous Exception) are not a problem as the new ones are subclass of the old, but the change in exception message means backwards incompatible interface change for anyone using this library programmatically.

This is not a critical problem - and I understand it simplifies the code - but it should be released as version 2.0.0 as instructed in CRL workflow at https://crl-devutils.readthedocs.io/en/latest/workflow.html?highlight=version#is-version-updated.

Comment (OneTorcha) [reply src/crl/rfcli/rfcli.py]: Ok, I bring those errors back if we figure out the path issue. Small but might cause some projects a lot of editing :)

**Comment by theodore86** [comment]
```console
rfcli -t targets/target9.ini test.robot 
/tmp/targets/target9.ini
==============================================================================
Test                                                                          
==============================================================================
Test 01                                                               test
Test 01                                                               | PASS |
------------------------------------------------------------------------------
Test                                                                  | PASS |
1 test, 1 passed, 0 failed
==============================================================================
Debug:   /tmp/rfcli_output/debug.txt
Output:  /tmp/rfcli_output/output.xml
Log:     /tmp/rfcli_output/log.html
Report:  /tmp/rfcli_output/report.html
(⎈|:N/A)vagrant@sandbox:/tmp$ rfcli -t target9 test.robot 
/tmp/targets/target9.ini
==============================================================================
Test                                                                          
==============================================================================
Test 01                                                               test
Test 01                                                               | PASS |
------------------------------------------------------------------------------
Test                                                                  | PASS |
1 test, 1 passed, 0 failed
==============================================================================
Debug:   /tmp/rfcli_output/debug.txt
Output:  /tmp/rfcli_output/output.xml
Log:     /tmp/rfcli_output/log.html
Report:  /tmp/rfcli_output/report.html
(⎈|:N/A)vagrant@sandbox:/tmp$ cat targets/target9.ini 
[target]
USERNAME=test
```

**Comment by theodore86** [comment]
> @theodore86 maybe, review could be nice and for some reason sourcing libs and resources need "./" after this change..
> 
> like libraries/own_lib.py needs to be modified ../libraries/own_lib.py
> 
> Just havent had time to figure out or is that a big problem even.

Well it seems that you have remove the functionality which sets the libraries, testcases and resources to ```PYTHONPATH```:

```python
            print("Environment: %s" % self.full_environment)
            print("Commandline: %s" % self.commandline)
        # pylint: disable=unused-variable
        for key, value in self.new_environment_variables.items():
            newpaths = value.split(os.pathsep)
            for path in newpaths:
                pythonpathsetter.add_path(path, True)
```

derives from:

```python
    def run(self):
        if self.help_only:
            self.parser.print_help()
            return 0
        if self.use_jybot():
            command = JybotCommand(self.get_robot_options(), self.robot_args, self.get_environment(),
                                   debug=self.rfcli_args.debug)
        else:
            command = PybotCommand(self.get_robot_options(), self.robot_args, self.get_environment(),
                                   debug=self.rfcli_args.debug)
```

```python
    def get_environment(self):
        return {} if self.rfcli_args.no_pythonpath else self._get_environment()

    def _get_environment(self):
        testcase_libs = [
            i[0] for i in os.walk(os.path.join(self.calldir, 'testcases'), followlinks=True)
            if re.search(os.path.join('.*', 'libraries$'), i[0])]
        path_list = [self.calldir, os.path.join(
            self.calldir, 'libraries'), os.path.join(
                self.calldir, 'resources')] + testcase_libs
        return {'PYTHONPATH': os.pathsep.join(path_list)}

````

The proposal is this one (restore the part that you have removed and replace the following lines):

```python
        for key, value in self.new_environment_variables.items():
            newpaths = value.split(os.pathsep)
            for path in newpaths:
                if not any(fnmatch.fnmatch(p, path) for p in sys.path):
                    sys.path.append(path)
```

You just need 2 lines of code (```import fnmatch```)

**Comment by OneTorcha** [comment]
add_path() has been removed from 6.1.1, hence this "hack"/"fix". And I was trying to get it work bot 6.1.1 and older versions. I'll try your proposal, looks good.

**Comment by theodore86** [comment]
> add_path() has been removed from 6.1.1, hence this "hack"/"fix". And I was trying to get it work bot 6.1.1 and older versions. I'll try your proposal, looks good.

```robot.pythonpathsetter``` is not part of robot.api, indeed seems like a "hack" but adding those 2 lines of code (check the code above) and removing the ```pythonpathsetter``` you are done, the original implementation "appends" each path in the ```sys.path``` if not exists.

**Comment by OneTorcha** [comment]
So its better to make new branch and make just those little changes so I wont mess everything up :)

**Comment by theodore86** [comment]
> ```python
> if not any(fnmatch.fnmatch(p, path) for p in sys.path):
>                     sys.path.append(path)
> ```

Yea those 2 lines shall do the work.

It works btw:) waiting for the new release!

**Comment by OneTorcha** [comment]
@vereszol @theodore86 There is  PR comments fix, part 2. I think I covered everything, if not, ups :)

**Comment by theodore86** [comment]
@OneTorcha do you need any assistance on the PR comments?

**Comment by OneTorcha** [comment]
> @OneTorcha do you need any assistance on the PR comments?

Hi, what you mean? Do I have to Close With comment all? :)

**Comment by theodore86** [comment]
> > @OneTorcha do you need any assistance on the PR comments?
> 
> Hi, what you mean? Do I have to Close With comment all? :)

When are you planning to release the rfcli?

**Comment by OneTorcha** [comment]
> > > @OneTorcha do you need any assistance on the PR comments?
> > 
> > 
> > Hi, what you mean? Do I have to Close With comment all? :)
> 
> When are you planning to release the rfcli?

Whenever @vereszol accepts this and pulls the trigger.

**Comment by theodore86** [comment]
> > > > @OneTorcha do you need any assistance on the PR comments?
> > > 
> > > 
> > > Hi, what you mean? Do I have to Close With comment all? :)
> > 
> > 
> > When are you planning to release the rfcli?
> 
> Whenever @vereszol accepts this and pulls the trigger.

Do we need someone else to approve this PR?

**Comment by OneTorcha** [comment]
> > > > > @OneTorcha do you need any assistance on the PR comments?
> > > > 
> > > > 
> > > > Hi, what you mean? Do I have to Close With comment all? :)
> > > 
> > > 
> > > When are you planning to release the rfcli?
> > 
> > 
> > Whenever @vereszol accepts this and pulls the trigger.
> 
> Do we need someone else to approve this PR?

Smaller commits (one huge was not inclusive enough :) ), some refactoring for CI stuff.

**Comment by theodore86** [comment]
Whats the issue with the PR? It takes ages to be merged, The maintainers are not aware of it?

**Comment by OneTorcha** [comment]
> Whats the issue with the PR? It takes ages to be merged, The maintainers are not aware of it?

Too many changes in one PR. Pipe is porked. Nobody cares. I use the fix daily with my own hack and I have done what I need to do so rest is on that merge pull request button :)

**Pull request closed without merging**